transaction/_transaction.py

```diff
@@ -138,6 +138,8 @@ def user(self):
     def user(self, v):
         if v is not None:
             self._user = text_or_warn(v)
+        else:
```

Comment:
No. Nor  do I think setting txn.user = 1.23 is a valid use case.  In general, we don't type check in Python.  The only reason for the original None check was that I'd misunderstood someone to have asserted that Zope code set to None, which it hadn't.  I can live with the None check, because we're waaaaaay past the point of diminishing returns here.